Legg til overgangsordning opphørstekster automatisk

UyQuangNguyen opened the pull request:
Favrokort: https://favro.com/organization/98c34fb974ce445eac854de0/1844bbac3b6605eacc8f5543?card=NAV-23433

<img width="849" alt="Screenshot 2024-11-27 at 00 06 42" src="https://github.com/user-attachments/assets/39a2233c-8962-4c65-b75f-925ba8b3a000">
<img width="1466" alt="Screenshot 2024-11-27 at 00 06 35" src="https://github.com/user-attachments/assets/52f4f387-7916-4b64-82f8-374136d8bec5">

Legger til auto valg av opphør begrunnelse + tekst om forskrift.

UyQuangNguyen committed:
Legg til begrunnelser for opphør med autovalg dersom behandling er overgangsordning

UyQuangNguyen committed:
Legg til begrunnelser for opphør med autovalg av disse dersom behandlingen er overgangsordning. Legger også til tekst om forskrift.

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk

thoalm commented on src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/vedtaksperiode/Opphørsperiode.kt:
```diff
@@ -165,16 +168,49 @@ private fun finnOpphørsperioderMellomUtbetalingsperioder(utbetalingsperioder: L
+) {
+    val behandlingKategori = vedtaksperiodeMedBegrunnelser.vedtak.behandling.kategori
+
+    if (behandlingKategori == BehandlingKategori.NASJONAL) {
```
Her kunne man kanskje vurdert en `when` for å forhindre at andre kategorier blir lagt inn og automatisk blir behandlet som EØS

UyQuangNguyen replied:
Godt poeng! ⭐

thoalm commented on src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/vedtaksperiode/Opphørsperiode.kt:
```diff
@@ -165,16 +168,49 @@ private fun finnOpphørsperioderMellomUtbetalingsperioder(utbetalingsperioder: L
         vedtak = vedtak,
         type = this.vedtaksperiodetype,
     ).also { vedtaksperiodeMedBegrunnelser ->
+        if (behandlingsÅrsak == BehandlingÅrsak.OVERGANGSORDNING_2024) {
```
Her kunne man også vurdert en `when` for å sørge for at man må ta stilling til denne logikken om man legger til en ny årsak

UyQuangNguyen replied:
Jeg lar denne stå med tanke på at logikken som utføres i if'n heter settOvergangsordningOpphørBegrunnelser om er tett knyttet til OVERGANGSORDNING_2024

thoalm commented on src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevUtil.kt:
```diff
@@ -179,3 +187,10 @@ fun String.tilLandNavn(landkoderISO2: Map<String, String>): Landkode {
+
+fun List<SanityBegrunnelse>.inneholderOvergangsordningBegrunnelser() =
+    this.any {
+        it.apiNavn == NasjonalEllerFellesBegrunnelse.INNVILGET_OVERGANGSORDNING.sanityApiNavn ||
```
Om dette ikke er kode som skal slettes kan man vurdere å lage en metode i `NasjonalEllerFellesBegrunnelse` enumen som sjekker om det er overgangsordning eller ikke

thoalm reviewed (approved):
Du kan vurdere de kommentaren jeg la inn, men ser ellers bra ut for min del 

Testene kommer i en senere commit

UyQuangNguyen committed:
Merge branch 'main' of github.com:navikt/familie-ks-sak into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Legg til tester på å legge inn opphørstekster

UyQuangNguyen committed:
Merge branch 'Legg_til_overgangsordning_opphørstekster_automatisk' of github.com:navikt/familie-ks-sak into Legg_til_overgangsordning_opphørstekster_automatisk

MagnusTonnessen committed:
Legg til kalkulert utbetalingsbeløp for ordinær andel i utbetalingsmåned for overgangsordning (#966)

### 💰 Hva skal gjøres, og hvorfor?

Det ble oppdaget en bug i behandlinger for overgangsordning der
utbetalingsoppdraget for overgangsordning overskriver ordinær
utbetalingsoppdrag i utbetalingsmåneden for overgangsordningen. Fikser
dette ved å inkludere `kalkulertUtbetalingsbeløp` for ordinær andel i
utbetalingsandelen til overgangsordning.

### ✅ Checklist
_Har du husket alle punktene i listen?_
- [ ] Jeg har testet mine endringer i henhold til akseptansekriteriene
🕵️
- [ ] Jeg har config- eller sql-endringer. I så fall, husk manuell
deploy til miljø for å verifisere endringene.
- [x] Jeg har skrevet tester. Hvis du ikke har skrevet tester, beskriv
hvorfor under 👇

### 💬 Ønsker du en muntlig gjennomgang?
- [ ] Ja
- [x] Nei

MagnusTonnessen committed:
Legg til kalkulert utbetalingsbeløp for ordinær andel i utbetalingsmåned for overgangsordning (#966)

Det ble oppdaget en bug i behandlinger for overgangsordning der
utbetalingsoppdraget for overgangsordning overskriver ordinær
utbetalingsoppdrag i utbetalingsmåneden for overgangsordningen. Fikser
dette ved å inkludere `kalkulertUtbetalingsbeløp` for ordinær andel i
utbetalingsandelen til overgangsordning.

_Har du husket alle punktene i listen?_
- [ ] Jeg har testet mine endringer i henhold til akseptansekriteriene
🕵️
- [ ] Jeg har config- eller sql-endringer. I så fall, husk manuell
deploy til miljø for å verifisere endringene.
- [x] Jeg har skrevet tester. Hvis du ikke har skrevet tester, beskriv
hvorfor under 👇

- [ ] Ja
- [x] Nei

Legg til tester på sjekk av overgangsordning begrunnelse

Legg til tester på sjekk av overgangsordning begrunnelse

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Legg til tester på sjekk av overgangsordning begrunnelse

UyQuangNguyen committed:
Merge branch 'Legg_til_overgangsordning_opphørstekster_automatisk' of github.com:navikt/familie-ks-sak into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Legg til tester på sjekk av overgangsordning begrunnelse

UyQuangNguyen committed:
Merge branch 'Legg_til_overgangsordning_opphørstekster_automatisk' of github.com:navikt/familie-ks-sak into Legg_til_overgangsordning_opphørstekster_automatisk

UyQuangNguyen committed:
Revert uante endringer

UyQuangNguyen committed:
Legg til flere tester

Legg til flere tester

UyQuangNguyen committed:
Merge branch 'main' into Legg_til_overgangsordning_opphørstekster_automatisk
